src/Illuminate/Filesystem/FilesystemManager.php

```diff
@@ -169,7 +169,8 @@ protected function callCustomCreator(array $config)
     public function createLocalDriver(array $config)
     {
         $visibility = PortableVisibilityConverter::fromArray(
-            $config['permissions'] ?? []
+            $config['permissions'] ?? [],
+            $config['visibility'] ?? Visibility::PRIVATE
```

Comment:
Can you add a new `directory_visibility` config option as well that falls back on `visibility`? That way people can fine-grain permissions. `directory_visibility` conforms with Flysystem's own new option for it: https://github.com/thephpleague/flysystem/blob/3.x/src/Config.php#L12

```suggestion
            $config['directory_visibility'] ?? $config['visibility'] ?? Visibility::PRIVATE
```

Reply:
Nvm, added this myself.